utils/cache.js

```diff
@@ -160,6 +160,11 @@ const cacheResponse = (options = {}) => {
         const oldSend = res.send;
 
         res.send = (body) => {
+          if (res.statusCode < 200 || res.statusCode >= 300) {
+            res.send = oldSend;
+            return res.send(body);
+          }
```

Comment:
from where in the code do we get this res.statusCode (middleware/controller/somewhere else)?

Reply:
```javascript
 /**
         * As we do not have data in our cache we call the next middleware,
         * intercept the response being sent from middleware and store it in cache.
         *  */
        const oldSend = res.send;


        res.send = (body) => {
    
          if (res.statusCode < 200 || res.statusCode >= 300) {
            res.send = oldSend;
            return res.send(body);
          }

          const cacheValue = {
            priority: priority,
            response: body,
            expiry: new Date().getTime() + minutesToMilliseconds(expiry),
            size: Buffer.byteLength(body),
          };
          pool.set(key, cacheValue);
          if (modelKey) {
            cachedKeys.addCachedKey(modelKey, key);
          }
          res.send = oldSend;
          return res.send(body);
        };

        next();
``` 

In our `cacheResponse` middleware we are just assigning a new method to `res.send` 

This will be called in controller using res.send or res.json or res.boom.. etc..

Reply:
> This will be called in controller using res.send or res.json or res.boom.. etc..

if this is being called in the controller and then all the code from the controller will be made then the db calls will be made. werent we suppossed to protect that?